packages/rest/lib/resource.cadl

```diff
@@ -110,3 +110,89 @@ interface ResourceInstanceOperations<TResource, TError> mixes ResourceRead<TReso
 interface ResourceCollectionOperations<TResource, TError> mixes ResourceCreate<TResource, TError>, ResourceList<TResource, TError> {}
 
 interface ResourceOperations<TResource, TError> mixes ResourceInstanceOperations<TResource, TError>, ResourceCollectionOperations<TResource, TError> {}
+
+interface SingletonResourceRead<TSingleton, TResource, TError> {
```

Comment:
I am not following the intended pattern.  I think of a singleton as a resource that has a literal value for a name parameter 
 ( that is ```/...path/name/default```  where default is the string literal.  I don't think that is what is being modeled here, so what is our conception of a singleton resource in this case? Why would this not be covered by simply allowing key to be applied to a property with a literal string value type?

Reply:
The QnA Langauge service does not follow that pattern for its singleton resources.  Check out this "synonyms" subpath with `GET` and `PUT` operations:

https://github.com/Azure/azure-rest-api-specs/blob/main/specification/cognitiveservices/data-plane/Language/stable/2021-10-01/questionanswering-authoring.json#L579-L673

I agree that we should also support the other approach, but that may already be possible with the existing resource interface design (you can apply `@key` to a property with a literal string type and it will be injected into the path directly).

Reply:
This seems really bizarre.  The body parameter for PUT is a pageable collection?  It seems to me that this would be much better modeled as a child resource, and if there is need for a bulk data upload, having this as a separate operation.  I am skeptical that this is a pattern we want to encourage.